### PR TITLE
[FLINK-32848][tests][JUnit5 migration] Migrate flink-runtime/registration tests to JUnit5

### PR DESCRIPTION
**Pull request by X-czh**
## What is the purpose of the change

Migrate flink-runtime/registration tests to JUnit5.

## Verifying this change

The change itself is for migrating UT.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by X-czh** [comment]
Hi @1996fanrui, could you review the change when you are free, much appreciated~

**Comment by X-czh** [comment]
Hi @FangYongs, could you review the change when you are free?

File: flink-runtime/src/test/java/org/apache/flink/runtime/registration/RegisteredRpcConnectionTest.java [hunk 1]
```diff
@@ -23,45 +23,39 @@
...
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Tests for RegisteredRpcConnection, validating the successful, failure and close behavior. */
-public class RegisteredRpcConnectionTest extends TestLogger {
+class RegisteredRpcConnectionTest {
```
Comment (FangYongs) [inline flink-runtime/src/test/java/org/apache/flink/runtime/registration/RegisteredRpcConnectionTest.java:41]: You may need to add `@ExtendWith(TestLoggerExtension.class)` for `RegisteredRpcConnectionTest` after removing `extends TestLogger`

File: flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationConfigurationTest.java [hunk 2]
```diff
@@ -21,20 +21,18 @@
...
 
 import java.time.Duration;
 
-import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests for the {@link RetryingRegistrationConfiguration}. */
-public class RetryingRegistrationConfigurationTest extends TestLogger {
+class RetryingRegistrationConfigurationTest {
```
Comment (FangYongs) [inline flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationConfigurationTest.java:32]: Same as above

File: flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationTest.java [hunk 3]
```diff
@@ -59,28 +54,28 @@
  * Tests for the generic retrying registration class, validating the failure, retry, and back-off
  * behavior.
  */
-public class RetryingRegistrationTest extends TestLogger {
+class RetryingRegistrationTest {
```
Comment (FangYongs) [inline flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationTest.java:57]: Same as above

**Comment by FangYongs** [comment]
Thanks @X-czh for contribution, just some minor comments

File: flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationConfigurationTest.java (hunk 2 above)
Comment (X-czh) [reply flink-runtime/src/test/java/org/apache/flink/runtime/registration/RetryingRegistrationConfigurationTest.java:32]: Hi, we reached agreement with Fan Rui and Matthias Pohl previously that we should not add @ExtendWith(TestLoggerExtension.class) explicitly for each class but rather enable it globally in Junit5 configs. See https://issues.apache.org/jira/browse/FLINK-32866

**Comment by FangYongs** [comment]
Thanks @X-czh , LGTM +1

**Pull request merged**